src/internal/packager2/layout/oci.go

```diff
@@ -78,11 +80,28 @@ func (r *Remote) Push(ctx context.Context, pkgLayout *PackageLayout, concurrency
 		descs = append(descs, desc)
 	}
 
+	// Sort by Digest string
+	sort.Slice(descs, func(i, j int) bool {
+		return descs[i].Digest < descs[j].Digest
+	})
+
 	annotations := annotationsFromMetadata(pkgLayout.Pkg.Metadata)
+
+	// Perform the conversion of the string timestamp to the appropriate format in order to maintain backwards compatibility
+
+	t, err := time.Parse(time.RFC1123Z, pkgLayout.Pkg.Build.Timestamp)
+	if err != nil {
+		// TODO: if we change the format of the timestamp, we need to update the conversion here
```

Comment:
Good callout and error checking.

NIT: Let's extract the `time.RFC1123Z` as a pkg const like `OCITimeFormat` or sensible alternative. Then this TODO can be provided as a doccomment, e.g.
<img width="697" alt="Screenshot 2025-04-10 at 15 54 08" src="https://github.com/user-attachments/assets/550c9594-835e-4a1c-94e1-827b246ab6bd" />